internal/fs/fs.go

```diff
@@ -1501,14 +1501,20 @@ func (fs *fileSystem) renameDir(
 	}
 	pendingInodes = append(pendingInodes, oldDir)
 
-	// Fetch all the descendants of the directory recuirsively
+	// Fetch all the descendants of the old directory recursively
 	descendants, err := oldDir.ReadDescendants(ctx, int(fs.renameDirLimit+1))
 	if len(descendants) > int(fs.renameDirLimit) {
 		return fmt.Errorf("too many objects to be renamed: %w", syscall.EMFILE)
 	}
 
-	// Create the backing object of the new directory.
 	newParent.Lock()
+	// Fetch a single descendant of the new parent
+	if newParentDescendents, err := newParent.ReadDescendants(ctx, 1); err == nil && len(newParentDescendents) > 0 {
```

Comment:
Here, the `newParent` refers to the parent directory of the target. For instance, in

```
mv /gcs/foo/bar/ /gcs/foo/baz/
```

both the oldParent and newParent are the directory `/gcs/foo/`. Therefore, we cannot abort the renaming if the newParent isn't empty.

Reply:
Sorry about this!  CI also should have caught this -- I opened #606 to prevent further regressions.

Reply:
@gaul No worries. Thanks for your contribution!